### ISSUE TITLE
HubDNS - Custom domains set within Hub at launch time do not appear to automagically work?!

### ISSUE THREAD

**Issue by JedMeister** [turnkeylinux/tracker#1508]
It's been reported that when launching a new server, custom domains (set within the "Launch new server" page and managed by the Hub) are not being created as they should be. Apparently tklapp.com subdomains do appear to work as they should...!

IIRC the DNS name is actually set from within the server itself (the Hub provides the info and then `hubdns` associated the FQDN with the server via the Hub). If my memory is correct, then in theory both should work... So perhaps there is some detail I'm missing...

I've tagged `hubdns` here but if it turns out that isn't related, we should remove it...

**Comment by JedMeister** [comment]
This has been confirmed by another user.

Furthermore, it appears that the issue is that whilst [HubDNS](https://github.com/turnkeylinux/hubdns) has been updated to python3, one of it's dependencies; [pycurl-wrapper](https://github.com/turnkeylinux/pycurl-wrapper) hasn't!

It seems how this issue occurred was that we updated pycurl-wrapper to python3 and tested (also py3 updated) HubDNS and all was well. Then at a later date during the v16.0 dev process, we rolled back the py3 update for pycurl-wrapper because it's also a [TKLBAM](https://github.com/turnkeylinux/tklbam) dependency (and we haven't yet ported that to py3). We completely overlooked the fact that HubDNS also depended on pycurl-wrapper.

The user who reported this noted that rolling back to a py2 version of HubDNS resolves the issue, but IMO a better option would be to publish a py3 pycurl-wrapper package. Perhaps we should just call it `py3curl-wrapper`?!

**Comment by JedMeister** [comment]
Apologies that it's taken me so long to circle back to this, but...

I have and have just uploaded 2 new packages to the `buster-testing` repo; an updated version of `hubdns` and `py3curl-wrapper` (a new python3 version of pycurl-wrapper - built from the [python3 branch of pycurl-wrapper](https://github.com/turnkeylinux/pycurl-wrapper/tree/python3)).

To install:

```
mv /etc/apt/sources.list.d/turnkey-testing.list.disabled /etc/apt/sources.list.d/turnkey-testing.list
apt update
apt upgrade hubdns
```

Then initialise and update:
```
hubdns-init YOUR_HUB_API_KEY YOUR.DOMAIN.COM
hubdns-update
```
(where `YOUR_HUB_API_KEY` & `YOUR.DOMAIN.COM` are your Hub API key and your desired domain name respectively)

[Note that currently this requires the TurnKey `buster-testing` repo enabled, ASAP I will move these packages to the `main` TurnKey repo]

**Comment by JedMeister** [comment]
This has now been moved to TurnKey Buster `main` repo so can now be installed with just:
```
apt update
apt install hubdns
```
This will be included in all future appliance releases (v16.1 release coming soon...). So I'm closing this now.
